controller/src/main/java/io/pravega/controller/server/eventProcessor/requesthandlers/SealStreamTask.java

```diff
@@ -59,6 +68,16 @@ public SealStreamTask(final StreamMetadataTasks streamMetadataTasks,
                         throw new TaskExceptions.StartException("Seal stream task not started yet.");
                     }
                 })
+                .thenCompose(x -> cancelTransactions(context, scope, stream)
+                        .thenAccept(cancelled -> {
+                            if (!cancelled) {
+                                // If transactions exist on the stream, we will throw OperationNotAllowed so that this task
+                                // is retried.
+                                throw StoreException.create(StoreException.Type.OPERATION_NOT_ALLOWED,
```

Comment:
we could log the scope and stream name here, right?

Reply:
I will add debug level log here as this is not an error case.. its just postponement scenario.